lib/Solver/Z3Builder.cpp

```diff
@@ -427,22 +428,25 @@ Z3ASTHandle Z3Builder::getInitialRead(const Array *root, unsigned index) {
 
 Z3ASTHandle Z3Builder::getArrayForUpdate(const Array *root,
                                          const UpdateNode *un) {
+  Z3ASTHandle un_expr;
+  std::vector<const UpdateNode *> update_nodes;
+  for (; un && !_arr_hash.lookupUpdateNodeExpr(un, un_expr);
+       un = un->next.get()) {
+    update_nodes.emplace_back(un);
```

Comment:
What's the point of `emplace_back` instead of `push_back` here?

Reply:
A good habit ;)

Reply:
If you prefer `push_back`, that is an easy enough change...